client/my-sites/controller.js

```diff
@@ -99,6 +99,10 @@ module.exports = {
 		}
 
 		if ( currentUser && currentUser.site_count === 0 ) {
+			context.store.dispatch( uiActions.setSection( 'sites', {
```

Comment:
Both this case and `renderEmptySites` do the following:

- dispatch with `hasSidebar: false`
- unmount the `#secondary` div

Maybe we should move both those actions to a new function (`removeSidebar`)?